src/pkg/reg/adapter/aliacr/openapi.go

```diff
@@ -0,0 +1,317 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aliacr
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/cr"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/cr_ee"
+)
+
+type repository struct {
+	Name      string
+	Namespace string
+	ID        string
+}
+
+type authToken struct {
+	user      string
+	password  string
+	expiresAt time.Time
+}
+
+// openapi is an interface that defines methods for interacting with an open API.
+type openapi interface {
+	// ListNamespace returns a list of all namespaces.
+	ListNamespace() ([]string, error)
+	// ListRepository returns a list of all repositories for a specified namespace.
+	ListRepository(namespaceName string) ([]*repository, error)
+	// ListRepoTag returns a list of all tags for a specified repository.
+	ListRepoTag(repo *repository) ([]string, error)
+	// GetAuthorizationToken returns the authorization token for repository.
+	GetAuthorizationToken() (*authToken, error)
+}
+
+type acrOpenapi struct {
+	client *cr.Client
+	domain string
+}
+
+var _ openapi = &acrOpenapi{}
+
+// newAcrOpenapi creates a new acrOpenapi instance.
+func newAcrOpenapi(accessKeyID string, accessKeySecret string, regionID string) (openapi, error) {
+	client, err := cr.NewClientWithAccessKey(regionID, accessKeyID, accessKeySecret)
+	if err != nil {
+		return nil, err
+	}
+	return &acrOpenapi{
+		client: client,
+		domain: fmt.Sprintf(endpointTpl, regionID),
+	}, nil
+}
+
+// ListNamespace returns a list of namespaces
+func (acr *acrOpenapi) ListNamespace() ([]string, error) {
+	var namespaces []string
+	var nsReq = cr.CreateGetNamespaceListRequest()
```

Comment:
Please revise them a little bit to make it more readable.

```
nsReq := cr.CreateGetNamespaceListRequest()
nsReq.SetDomain(acr.domain)

nsResp, err := acr.client.GetNamespaceList(nsReq)
if err != nil {
	return nil, err
}
```

Reply:
done